### DISCUSSION TITLE
AzureMonitor: Don't maintain whitelist of supported metric namespaces

### DISCUSSION THREAD

**Discussion by joshhunt** [grafana/grafana#41982] [Data Sources]
Having a manually maintained list of [`supportedMetricNamespaces`](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/supported_namespaces.ts) that controls the metric namespaces that can be queries through Azure Monitor metrics leads to a maintenance burden and feature requests from missing services, such as #28733.

Needs investigating, but we should either remove this list (don't filter on our end) or use an API to get supported metric namespaces. Will need to investigate how National Clouds plays into this.

**Comment by M0ns1gn0r** [comment]
It would be nice to have this implemented. I need the `Microsoft.SignalRService/SignalR` metric, which is not on the list, and I don't know how to proceed. Should I create a separate issue with this request? 

So realistically, what's the chance that this issue gets resolved any time soon? Is it really planned or just something nice to have in the far future?

**Comment by joshhunt** [comment]
Hey @M0ns1gn0r - we can't commit to timelines, but this is something that is of a medium priority that we do want to get done.

For now, PRs are welcome for metric namespace additions until this issue is address specifically and/or you have the option of running Grafana from source to get the change in before the release. https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/supported_namespaces.ts#L3

**Comment by joshhunt** [comment]
Could be useful https://docs.microsoft.com/en-us/rest/api/monitor/metricnamespaces/list

**Comment by kostrse** [comment]
Probably related: #33566

**Comment by joshhunt** [comment]
Thinking about this further, if there is no simple way to filter metrics/resources from the the Azure APIs or ARG to only those supported by Metrics or Logs, I think I would just prefer to not filter them at all, and if a query is executed to an unsupported resource then we display a friendly error message to explain this.

I think that is a better experience compared to currently where resources will just be missing.

**Comment by andresmgot** [comment]
Moved to an issue: https://github.com/grafana/grafana/issues/50239 keep the discussion there if needed!
